### DISCUSSION TITLE
Remember track overlays in import route

### DISCUSSION THREAD

**Discussion by GuG1050** [devemux86/cruiser#431] [Q&A] [answered]
I import a .kurviger file overlay + routing. Both is displayed. This morning I discovered that when I close the app and then reopen only the routing is displayed. The overlay has disappeared. I am shure this was different (and better) before. Can you revert to the previous behavior of the App? My Version ist 3.0.37

**Comment by devemux86** [comment]
@GuG1050 thanks for the report.

Now the routes or tracks we want to keep can be saved in favorites and displayed at any time,
each with a different color and visibility options.

<img src="https://github.com/devemux86/cruiser/assets/3484020/95152bf5-dac6-492d-a99e-4feae8d3c5ab" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/d0af3f6a-548e-4928-b181-5205730b47ab" width="50%"/>

**Reply by WolfgangZ42** [reply]
> tracks we want to keep can be saved in favorites

That's true - but: Then I would have to transfer every track to favorites, even though it is not one of my favorites. Actually, I just want to leave the app and then continue riding with it the next morning. So may I ask: can't we keep the track when we switch off the app, independent of the favorites function?

**Reply by devemux86** [reply]
@WolfgangZ42 it's understandable, we could do something like that.
It just becomes complicated to manage. :slightly_smiling_face:

**Reply by WolfgangZ42** [reply]
Perhaps we could have a "current track" which is cleared only manually or by importing another track (via kurviger- or gpx-file) - as before 3.0.37?. The display of favorite tracks is managed with "my places" anyway.

**Reply by bios4** [reply]
> So may I ask: can't we keep the track when we switch off the app, independent of the favorites function?

> @WolfgangZ42 it's understandable, we could do something like that. It just becomes complicated to manage. 🙂

Are there still plans to make this possible again that the overlay of a loaded GPX/Kurviger route remains on the map after the app is closed/reloaded @devemux86?

I would highly appreciate it...

**Reply by devemux86** [reply]
> Are there still plans to make this possible again that the overlay of a loaded GPX/Kurviger route remains on the map after the app is closed/reloaded

It is already in my list for implementation. :slightly_smiling_face:

**Comment by GuG1050** [comment]
Ahh. Much better than before 🙂. I should spend even more time playing with the app.

**Comment by devemux86** [comment]
Remember overlays during import route will be available in Cruiser **5.2**.

- These imported overlays will be stored in "My collection"
- Their color, visibility can be adjusted like the other tracks
- They can be deleted individually or all with "Clear" menu

**Comment by devemux86** [comment]
Available for testing in:
- https://github.com/devemux86/cruiser/discussions/690

**Comment by devemux86** [comment, marked as answer]
- Remember track overlays in import route is available in https://github.com/devemux86/cruiser/discussions/701

**Comment by airald** [comment]
I think that's very useful to have the track visible during navigation. What I don't know how to do it is clear the lines that joins the points of the route. I can just hide the track or clear ALL overlays. Is is possible see the route calculated and the track? in the example, just delete the green line.

![Screenshot_2025-01-08-22-01-55-983_gr talent cruiser](https://github.com/user-attachments/assets/55234388-03b8-43c4-b718-046b45c88764)
![Screenshot_2025-01-08-22-02-41-105_gr talent cruiser](https://github.com/user-attachments/assets/ed9c3227-3401-4529-826e-5b098c7902db)

**Reply by devemux86** [reply]
@airald This green line should the GPX `rte` route connecting the `rtept` route points.
(it is cleared if restart the app or clear overlay)

Probably we do not need it and we should only overlay the GPX `wpt` waypoints and `trk` track?

**Reply by WolfgangZ42** [reply]
> Probably we do not need it 

This green rte line (in kurviger.com it even has the same color as the trk line!) is for me just disturbing, nothing else. I have often wondered who the hell needs this connection line. But: there may be other opinions ...

**Reply by airald** [reply]
If I delete the overlay, the green line that connects the route points disappear. I don't need this line. For me it's something that introduces "visual noise" when importing a route.

**Reply by GuG1050** [reply]
For me this green line is useless. Can be removed.

**Reply by DolomitesTom** [reply]
Not tested yet but don't need too.

**Reply by devemux86** [reply]
> This green line should the GPX `rte` route connecting the `rtept` route points.

It will be removed in the next update.
